workflow/serializers.py

```diff
@@ -160,6 +196,40 @@ class Parameter(ModelSerializer):
         required=False
     )
 
+    def update(self, instance, validated_data):
+        raise_errors_on_nested_writes('update', self, validated_data)
+        info = model_meta.get_field_info(instance)
+
+        # Simply set each attribute on the instance, and then save it.
+        # Note that unlike `.create()` we don't need to treat many-to-many
+        # relationships as being a special case. During updates we already
+        # have an instance pk for the relationships to be associated with.
+        for attr, value in validated_data.items():
+
+            # THIS NEEDS TESTING
+            if attr in info.relations and info.relations[attr].has_through_model:
+                for related_instance in value:
+                    through_class = info.relations[attr].model_field.rel.through
```

Comment:
It would be helpful to include a link to some site with information about dealing with model internals like this, at the very least. I've never seen any of these functions or fields in my entire life, and I'd imagine I'm not alone in that

Reply:
I don't know of any. I used the source to figure this out

Reply:
can you link to the relevant parts of the source that you used?